Cql/Cql.Runtime/DefinitionDictionary.cs

```diff
@@ -485,20 +486,49 @@ orderby g.Key
         internal class Tag
         {
             public string Library { get; }
+            public string Definition { get; }
+            public Type[] Signature { get; }
+            public string Value { get; }
+            public string Name { get; }
 
             public Tag(string library, string definition, Type[] signature, string name, string value)
             {
                 Library = library;
                 Definition = definition;
-                Value = value;
-                Name = name;
                 Signature = signature;
+                Name = name;
+                Value = CleanValue(value);
             }
+            
+            //clean the 'Value' field
+            private static string CleanValue(string value)
+            {
+                if (string.IsNullOrEmpty(value)) return value;
+
+                char[] result = new char[value.Length];
+                int resultIndex = 0;
+                char space = (char)32;
+                char prevChar = space; 
+
+                foreach (char currentChar in value)

```

Comment:
This algorithm is hard to read for me. It looks like it tries to convert (consecutive) whitespace into a single space, but if that is the case, I think there is a bug in it (inserting a space when seeing a whitespace char other than space does not change prevChar to "space", so we'll still insert whitespace the next time?).  In any case it's good to document what to "clean the field" means!

Reply:
Thanks good eye on the bug. I fixed it and also made the setup a little more clear by using trim (which is still fast and also cleans up entirely empty annotations of whitespace characters). I opted for this manual process because of performance - a replace over and over was pretty slow by like X times its used, because it goes back through the string over and over.

Reply:
To be perfectly honest, because this code is just run at compile time (so once, when you are preparing the measure), it is not time critical as it is not run every time the measure runs.  So I would just do this:

```
var input = "Hello \r\n there, how are  you\tnowadays?";
var output = Regex.Replace(x, @"\s+", " ");
```

What do you think about that?